### PR TITLE
docs: add tapps meeting 2019-11-11 minute

### LINKED ISSUE #77: Next Tapps Meeting
Hey folks (@dhis2/team-apps & @amcgee)

There are several things we need to discuss, so I think I'll just write down what I currently have in my head (@varl feel free to take over this issue and rename title and change the description in any way you see fit)

## 1. How to proceed with the new Maintenance app

I've gotten quite far and reached a state where I don't want to continue working alone on this.
There are too many opinions and people involved.
So I think we need to talk about the next steps and prioritization. @varl mentioned we'll get some time for the new app this autumn but I don't know if that's still valid, how much time we can spend on it and how we should prioritize it in regards to other issues on our board. ([Link to the repo](https://github.com/dhis2/mar-y) & [Link to the JIRA epic](https://jira.dhis2.org/browse/TECH-223)). There is one open issue which I think is worth reading through. 

## 2. Error reporting

This is a cross-team effort, but we need to get it started somehow. I've porposed we (the devs) come up with an initial proposal so we have something to talk about.
It'll be quite hard to get all devs involved in this at once, so I think we (and @amcgee, who created the first JIRA issue about this) should come together and talk (ideally prepared, I've written down my thoughts here: #76)

## 3. Positioning engine (internal vs external vs very simple solution)

@cooper-joe has created a PR with features. I tried to improve some things from a dev perspective:
* Joe's PR: dhis2/ui-core#522
* My improvements: dhis2/ui-core#527

We can talk about his proposals and how we should act based on them

## 4. Remotes & remote culture

Lately I feel quite disconnected from dhis2 and I guess I'm not the only one. We have more or less properly working communication between remotes (due to the nature of working remotely). But it feels more and more demotivating that dhis2 itself it not really trying to adopt a remote culture.

I think it's crucial for dhis2 to hire remotes as we're understaffed even with the remotes we currently have. Which is why I think we should put more pressure on making it more remote friendly.

* I have no idea what's happening in Oslo (discussion wise).
Us remotes use GitHub actively, especially discussing stuff in `dhis2/notes`, but the lack of participation of others is not good (also from a functional cross team workflow perspective)
* It normally requires some waiting time if remotes need information
Asking a question on the `#webapi` channel sometimes takes a few attempts before being answered, sometimes there's no response whatsoever. It feels like we have to be pro-active all the time

I know this is not intentional and I also don't really know what we can do to change it. But we should address it and talk about possible solutions (hiring more BE remotes? hiring a remote team coach? Enforcing certain good practices like meetings where everyone uses his/her own computer? Enforcing discussions taking place on GitHub (no idea if that's even possible)? Use some work time to organize some kind of social online event that's entirely unrelated to work?)
This should be addressed in a very constructive way without any shaming at all! I'm not trying to criticize anyone, just the historically-explainable structures in place.

## 5. Documentation of our libraries

I'll just quote @amcgee here:

> And want to reiterate the Tutorial / How-To Guides / Reference / Conceptual Explanation documentation quadfecta, we should have all 4

Assuming that all of us think that this is important, we need to craft a plan how to proceed.
I've already proposed the following:

> I think we should create an epic, then tasks, then assign them
> Maybe have a meeting beforehand to align what part should document what and in what way

### PR DESCRIPTION
**Pull request by varl**
Closes #77.

### CODE REVIEW

File: decisions/2019/11/tapps-meeting-remote-errors-etc.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+- Document what the point of the dev portal is on the first page (better landing page).
+
+## General thoughts for future discussions
+
+- How do we keep ourselves accountable to our decisions?
+
+- Quarterly team focus meeting for contextualising our work
+
+- Roadmaps for the entire app platform (includes ui libraries, etc.)
```
Comment (HendrikThePendric) [inline decisions/2019/11/tapps-meeting-remote-errors-etc.md]: I don't disagree with the term "roadmap" and think defining roadmaps could be a useful approach for getting a better grip on planning. However, I'd like to emphasise another term that I mentioned during the meeting, which was Minimum Viable Product. IMAO this is the part that we are missing ATM. We try to do too much in the first iteration and should initially define things in terms of "need to have" and "nice to have" (and focus on need to have only for the MVP). If we get the MVP out of the door more quickly, and it is actually a well-rounded product, we can build from there, and start getting meaningful feedback from users/developers at a much earlier stage....

Comment (HendrikThePendric) [reply decisions/2019/11/tapps-meeting-remote-errors-etc.md]: To make my point above a little bit more concrete:
- Yes roadmaps are good
- Defining an MVP should be step 1 in the roadmap
- Building the MVP should be step 2
- Everything else can be done later

Comment (varl) [reply decisions/2019/11/tapps-meeting-remote-errors-etc.md]: Right. A roadmap in this situation is a tool to help define **needs** vs **wants**. Additionally a roadmap also provides context for **why** we are building a feature and **when** we should do so (what/why/how/when).

If we have a roadmap even for the first version decided, then it makes it easy to say what the scope or MVP is.

It is just a personal preference that I avoid the term MVP and instead talk about scope.

Comment (HendrikThePendric) [reply decisions/2019/11/tapps-meeting-remote-errors-etc.md]: 👍 As long as your definition of roadmap contains a separation between need-to-have and nice-to-have then it's all good with me.

File: decisions/2019/11/tapps-meeting-remote-errors-etc.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+When it comes to improvements and functional changes to the Maintenance
+application, it is a bit stickier, as any changes to the Maintenance
+application triggers a lot of work related to updating training materials and
+training for people. We should gather requirements and write specs for changes
+we think would be inline with our way of work.
+
+Input from the field from Austin:
+
+- Import/Export app cannot import GML
```
Comment (Birkbjo) [inline decisions/2019/11/tapps-meeting-remote-errors-etc.md]: You mean Maintenance App here, right?

Comment (Mohammer5) [reply decisions/2019/11/tapps-meeting-remote-errors-etc.md]: I think it's actually: Import/Export app can't import polyglyphs (is that the right term?) (which is a concern @amcgee raised, if I'm not mistaken)

Comment (Birkbjo) [reply decisions/2019/11/tapps-meeting-remote-errors-etc.md]: The problem that was raised in the meeting was that the Mainteanance app does not support selecting or uploading polygons for organisation units. Only manually entered coordinates are supported. That's why the import/export app is used to import GML/polygons.
I agree that this is something that should be supported in the maintenance app, but it's pretty complex if you want maps support, so this is a feature that would benefit greatly from cross-team cooperation with the GIS team. But we will probably not introduce this until the refactor is feature-complete.

Comment (Mohammer5) [reply decisions/2019/11/tapps-meeting-remote-errors-etc.md]: Ah I see, my bad. You're right and I agree with your suggestion

Comment (varl) [reply decisions/2019/11/tapps-meeting-remote-errors-etc.md]: Good catch, @Birkbjo! My notes were inconclusive and I had this under the wrong section. I'll update.

Comment (varl) [reply decisions/2019/11/tapps-meeting-remote-errors-etc.md]: 933100c

File: decisions/2019/11/11-tapps-meeting-remote-errors-etc.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+- Build a library so that applications can report various metrics
+  that get visualised in the usage analytics app.
+
+- Extend the APIs for the usage analytics to enable additional
+  metrics (requires backend work).
+
+- Collect the data at the instance level and iteratively add more
+  features. "Send data to DHIS2 Core Team" can be a later feature.
+
+Next step is to schedule a meeting to set up a proposal to anchor the
```
Comment (amcgee) [inline decisions/2019/11/11-tapps-meeting-remote-errors-etc.md]: > Next step is to schedule a meeting to set up a proposal to anchor the
discussion across teams in a context.

A more concrete step than meeting-to-propose-to-discuss might be nice , like "(1) someone (@Mohammer5 or myself?) creates a proposal for tasks and timeline, (2) we meet to approve or modify it"

Comment (Mohammer5) [reply decisions/2019/11/11-tapps-meeting-remote-errors-etc.md]: Agreed. 

OT: @amcgee what do you think about us meeting and creating the draft by discussing ideas? We could just prepare ourselves and join in a hangouts call and discuss it & create a proper proposal? I guess you're more familiar with dhis2 and the already existing analytics stuff, I might add some stuff from my new relic experience and from what I was thinking about so far.

Comment (amcgee) [reply decisions/2019/11/11-tapps-meeting-remote-errors-etc.md]: Sounds great to me

Comment (varl) [reply decisions/2019/11/11-tapps-meeting-remote-errors-etc.md]: Sounds like you already scheduled the meeting then, I'll update the minute and merge it. Thanks guys.

**Pull request merged**